### DISCUSSION TITLE
Keyboard shortcuts keep failing to get loaded

### DISCUSSION THREAD

**Discussion by kendew** [lxqt/lxqt#2169] [Q&A] [answered]
I've been using LXQt for years as one of my favorite DEs next to Sway and Openbox on Arch Linux and I never encountered this problem until recently.  I can't get LXQt to remember keyboard shortcuts consistently on login, either my own or the defaults.  Sometimes it does and sometimes it doesn't.  I've tried eliminating ~/.config/kglobalshortcutsrc and ~/.config/lxqt/globalshortcuts.conf to return to defaults, but the problem persists, both on a NUC8i5BEH and Minisforum UM700.
When this problem manifests, both of the above files look perfectly fine, but Preferences --> LXQt Settings --> Shortcut Keys is completely empty, with no possibility to add new shortcuts (button greyed out).  
Has anyone else been having this problem?  It's new to me, which makes me wonder if it's produced by a recent update.
Kendew

**Comment by ssrublev** [comment]
I see something like this in Fedora. It's in "Session Settings" and then "Basic Settings". There are "LXQt Modules". There is a module called "Global Keyboard Shortcuts". It can't start sometimes instantly after reboot. Sometimes it is started by itself after some time. I also may try to start in manually several times. After several tries, it can start. List of shortcuts is empty when service is not started. When started, the list is present and daemon works normally. I recently started using LXQt, not long ago.

**Reply by tsujan** [reply]
→ https://github.com/lxqt/lxqt-globalkeys/issues/247

In short, it started after libx11 was upgraded to 1.7.3. Nothing has changed in `lxqt-globalkeys`.

**Reply by kendew** [reply]
Thank all for replies.  Especially for the promptness.  
Yes, reading the posts at the page tsujan linked to I see I'm not the only person experiencing this.  It must be frustrating for the developers to have this libx11 problem, something they have no control over.  For the moment I am using stefonarch's workaround and it seems to be working.  We will see with time because, as I mentioned, the issue is intermittent for me, sometimes appearing on login and sometimes not.  
Just as extra information, on an NUC8 I have an elaborate openbox setup next to LXQt, whereas on the UM700 I have only LXQt (except for the openbox that Arch installs by default).  One machine is Intel and the other AMD, so that also is not a factor for me.  
Except for ssrublev's experience above, mostly Arch or Arch based distros were mentioned.  
I set up Linux on computers for people and focus mainly on LXQt.  Quite a few of these users are Linux newbies so it's a problem when something goes wrong.  Reading [tsujan's linked page ](https://github.com/lxqt/lxqt-globalkeys/issues/247 ) I am really impressed by the detailed responses and concern for this issue.  It makes me feel my choice of LXQt is well placed.  
Still the curious issue remains as to why some of us are experiencing this issue and others like tsujan are not experiencing it at all??  I wonder if other DEs are experiencing similar libx11 problems?  Would help if they were as the libx11 issue might get fixed all the sooner...
The only difference I'm making to stefonarch's workaround is to make lxqt-globalkeyshortcuts-fallback.desktop a local file.  That way users can easily get rid of it without needing sudo privileges.  I don't suppose /usr/local/bin/globalkeyscheck will present a problem remaining where it is once this issue has been resolved.

**Reply by ssrublev** [reply]
Right now I tend to think that in Fedora, IF I JUST DON'T TOUCH PC FOR A WHILE AFTER REBOOT, the daemon begins to work by itself finally. Yesterday nvidia driver was updated from 495.44 to 495.46. My system acted strangely and I suggested that "xorg-nvidia" and "libx11" may affect each other. But I have no direct sympthoms, just strange behavior (brought me to Mesa driver one time with startup problems).

**Reply by stefonarch** [reply]
> The only difference I'm making to stefonarch's workaround is to make lxqt-globalkeyshortcuts-fallback.desktop a local file. That way users can easily get rid of it without needing sudo privileges.

Nice to hear that it works, and yes, for average Joe or Jane local files are a better solution.

**Comment by ssrublev** [comment]
Well, may somebody can help me why can't I assign keys Win+E for File Manager and Win+D for Show Desktop (like in Windows). This is in lxqt-globalkeys in Fedora.

Win+W and Win+S are allowed. Ctrl+E and Ctrl+D work too. But not Win+E and Win+D. No reaction while listening 10 seconds to keys.

**Reply by stefonarch** [reply]
Checked if they are set in the WM settings already?

**Reply by ssrublev** [reply]
Yes this looks like is related to keyboard bindings from MATE. But simple disabling MATE bindings don't help instantly. When I start MATE configuration applications from LXQt, it becomes buggy. A new buggy entry in "System Tray" appears — probably MATE's keyboard indicator that has buggy images instead of indicator.

Right now, I started "mate-keybinding-properties" from LXQt and tried to reassign "Win+E". Strange things begin to happen. "lxqt-globalkeys" list disappeared after that, and LXQt globalkeys stopped working. On Win+E, now Openbox warning appears saying: "Failed to execute child process "kfmclient" (No such file or directory)".

Well, but now feel I should keep trying more =)))

**Reply by tsujan** [reply]
Global shortcuts can be assigned only by one daemon/WM, which "consumes" them, so to speak.

**Reply by Faehnchen** [reply]
@ssrublev Just signed in to give my two cents. Hope it helps...

I remember I had this same issue on my new Lubuntu20.04 installation two weeks ago. For me the 'Show desktop'-entry didn't work as well. I binded it with Super_L (Windows) key as in my previous Lubuntu20.04-install on a different machine and I did recall from my windows-days. The first indicator was that the text of the path in the Keybindings GUI was in _italic_ , different than in all the other entries. Then I checked globalkeyshortcuts.conf. The entry for this keybind was out of the running order with the other entries, indicated by the number behind the bind in the [keybind.number]-title block, see:

[Super_L.42]
Comment=Show desktop
Enabled=true
path=/panel/showdesktop/show_hide 

There were no conflicts with Openbox keybindings, so I moved it back in the right order and saved but it still didn't work.
Then I cloned the entry and put it in order since you can't change 'Client' entries in the GUI. It now looked like this:

[Super_L.41]
Comment=Schreibtisch anzeigen
Enabled=true
path=/panel/showdesktop2/show_hide

Now it worked although beeing a double-entry. Only difference is the '2' after the command. Don't remember how it got there, but it works on all desktops, not just #two. Not knowing what did the trick felt a bit "dirty" :D.

I noticed that the globalkeyshortcuts GUI reshuffles the entries from time to time in a pattern that is obscure to me but certainly is for a reason. Perhaps something broke there.
However I hope that helps in yur case.

Apologies for my bad wording and the cumbersome answer. I'm not a dev, just an end user who after some irritations after switching from LXDE now fell in love with the LXqt desktop, althoug in Lubuntu 20.04 it appears still a bit "unfinished". After tinkering and learning about all the config files I now - after over ten years - finally get a grasp about what a good DE can do. I even started to use the terminal confidently :D. Many thanks to the devs! I would like to contribute at least with some user-feedback in the future. :)

**Reply by tsujan** [reply]
@Faehnchen
Welcome to LXQt Discussions! And thanks for sharing your experience and helping other LXQt users! That's what Discussions are for.

I understand that your comment was a reply to @ssrublev's last comment. You may already know what follows but just for the sake of clarity:

 * The LXQt you might have seen in Lubuntu20.04 should be very outdated. In fact, the LXQt of Ubuntu 21.10 is outdated too. LXQt doesn't have long-term support; only its latest version is supported (currently 1.0.0).
 * The main problem that's mentioned on this page is of a totally different nature and is related to a recent X11 update.

**Reply by Faehnchen** [reply]
Thank you! Yes I'm now aware of but I just managed to read the X11-Thread after my reply, sorry! I'm only a regular user for many years and  this was like beeing witness of a emergency-surgery on the open heart, very impressive :D. The fact that you guys responded to this issue so quickly and vigorously is for me a sign that this project is indeed in good hands. I really hope that you get more devs on board and keep the philosophy of lxqt up!
Actually I was on the verge of installing Manjaro these days and go through the process of implementing lxqt. Would have probably run into the same problem, so I wait a bit and watch you guys ;). I likely will also update to the new Lubuntu LTS version in april. Hope it's fixed until then. Is there a communication between you and the Lubuntu-devs on these issues or do they just grab what's there and slap it on their ISO? I'm still very content with 20.04 so far and they did a good job overall to make it very usable (minor imperfections here and there)... just courious.

**Reply by tsujan** [reply]
> Is there a communication between you and the Lubuntu-devs on these issues...

They may forward some reports from Launchpad, especially when they feel that the issues are present in the latest LXQt. We mostly rely on the reports by users of rolling distros (like Arch, Manjaro, OpenSUSE Tumbleweed,...) because they have the latest stable release, but we've also received good reports by Ubuntu and Fedora users. Apart from that, LXQt is developed independently of specific distros.

**Reply by stefonarch** [reply]
There is no real issue using arch or any arch-based distro - 2 easy workarounds are enough  atm ;)
FYI there is an "unofficial!"  iso https://github.com/lxqt/lxqt/discussions/2126

Thanks for your feedback!

**Reply by Faehnchen** [reply]
Thank you guys! Yes, I will try that. I'm still learning and just slowly gaining confidence, haha.

**Reply by ssrublev** [reply]
I am trying to understand how "Global Actions Manager" works together with other WM/DE's components which intercept keys from each other. MATE is outdated/buggy towards GTK and brings some mess (I am used to MATE apps and gave them installed). 

I see that in "Global Actions Manager" there are special instructions for launching commands as "/pane/[command]" (Type "Client"). Are there instructions about these "/panel/" commands? I have play/pause buttons which are not mentioned in "Global Actions Manager". If I want to assign "play" key to play media, I enter shortcut for "XF86AudioPlay". What "/panel/" command should I use. I have found that I can use "playerctl" command. But how is should be in LXQt originally?

**Comment by tsujan** [comment, marked as answer]
`lxqt-globalkeys` 1.0.1 has been released. It includes the fix (or the workaround, depending on the interpretation of the problem). When it comes to your distro, please remove any workaround that you may have used.

**Reply by ssrublev** [reply]
Is there a way to get it implemented in current Fedora? Are of these people: https://fedoraproject.org/wiki/Changes/LXQt someone who can help?

**Reply by tsujan** [reply]
This question should be asked from Fedora maintainers. I don't think they read these comments.

**Reply by ssrublev** [reply]
I have asked here: https://bugzilla.redhat.com/show_bug.cgi?id=lxqt

**Reply by ssrublev** [reply]
No answer there. So I re-asked at the "fedora-lxqt" maillist:  https://lists.fedoraproject.org/archives/list/lxqt@lists.fedoraproject.org/thread/SFVUXZ2WOLT4LUXXBYEEBEU7KG4RUYYF/

**Comment by nick-theboatman** [comment]
Maybe the same issue, maybe not... just upgraded from Ubuntu 18.04 and LDXE to Ubuntu 20.04 and LXQt. Bound to work, then...
For reasons that don't really have a good reason (they just "are") with LXDE / LXQt I use 2 panels, 1 top and 1 bottom. I place the desktop switcher on the top and have 12 desktops. With LXDE I would  configure the desktop count (as 12) in ~/.config/openbox/lxde_rc.xml (look for <desktops>). xde_rc.xml now redundant obvs... Using obconf configures the desktop count fine but this is non-persistent and doesn't "happen" on boot (the desktop count stays at 2). Right clicking on the desktop switcher and configuring does not allow changing the count. Using App menu / Preferences / LXQt settings / LXQt Config Centre and going to System Settings and using "workspaces" also gets access to the config... the names are volatile but the count remains stable. Displaying numbers at least gives a display of something. Maybe I am imagining it but it looks like there are multiple config files and these are not being read properly.. Sorry to have not going into this further at the moment.

Improvement request: with LXDE these desktop views (in the switcher) had minature versions of what was on the desktop, so you could see where you were meant to be headed... The LXQt version offers only numbers or labels... bit of a backward step in my (subjective) view. Any chance of bringing back the minature views?

**Reply by stefonarch** [reply]
>  going to System Settings and using "workspaces" also gets access to the config.

`Systemsettings` sounds like the KDE ones, and workspaces are for Kwin there,

If your using openbox the file that counts is `~/.config/openbox/rc.xml`
